### PR TITLE
[examples] Add ElasticSearch example

### PR DESCRIPTION
**Pull request by aravindputrevu**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

File: examples/with-elasticsearch/pages/index.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+          href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Powered by{' '}
+          <img src="/vercel.svg" alt="Vercel Logo" className="logo" />
+        </a>
+      </footer>
+
+      <style jsx>{`
```
Comment (leerob) [inline examples/with-elasticsearch/pages/index.js:71]: Could this use CSS Modules instead like create-next-app?

File: examples/with-elasticsearch/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+If you want to learn more about Elasticsearch, visit the following pages:
+
+- [Elastic Stack](https://https://www.elastic.co/products)
+- [Elastic Documentation](https://elastic.co/docs)
+
+## Deploy your own
+
+Once you have access to the environment variables you'll need, deploy the example using [Vercel](https://vercel.com?utm_source=github&utm_medium=readme&utm_campaign=next-example):
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-elasticsearch&project-name=with-elasticsearch&repository-name=with-elasticsearch&env=ESS_CLOUD_ID,ESS_CLOUD_USERNAME,ESS_CLOUD_PASSWORD&envDescription=Required%20to%20connect%20the%20app%20with%Elasticsearch)
```
[the remark this answers is not in the document]
Comment (aravindputrevu) [reply examples/with-elasticsearch/README.md:14]: Thanks for suggesting this, I don't know this. I'd need to take this to the team for further consideration. Maybe for now, can we merge this one without integration?

[the remark this answers is not in the document]
Comment (leerob) [reply examples/with-elasticsearch/README.md:14]: Yeah, not blocking, just an idea :)

File: examples/with-elasticsearch/pages/index.js (hunk 1 above)
Comment (aravindputrevu) [reply examples/with-elasticsearch/pages/index.js]: @leerob moved the CSS into modules just like create-next-app.

**Comment by aravindputrevu** [comment]
Please let me know if there are any changes needed further.

**Review by leerob — DISMISSED** [review]

**Comment by aravindputrevu** [comment]
@leerob Could you please help me with what's going wrong with the example check? I've definitely staged all the commits!

**Comment by leerob** [comment]
There's a few failing checks that will need to be addressed, and then it's good to go! E.g. linting

**Review by leerob — DISMISSED** [review]

File: examples/with-elasticsearch/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (leerob) [inline examples/with-elasticsearch/.eslintrc.json:1]: I don't see eslint in the package.json, should this be removed?

Comment (aravindputrevu) [reply examples/with-elasticsearch/.eslintrc.json:1]: I think this got added while setting up `create-next-app`. Let me remove it!

**Review by leerob — APPROVED** [review]

**Pull request merged**